server/shared/storage/storage.controller.js

```diff
@@ -17,36 +17,37 @@ function getUrl(req, res) {
   return getFileUrl(key).then(url => res.json({ url }));
 }
 
-async function upload({ file, body, user }, res) {
+async function upload({ file, body, user, repository }, res) {
   const { name } = path.parse(file.originalname);
   if (body.unpack) {
     const timestamp = fecha.format(new Date(), 'YYYY-MM-DDTHH:mm:ss');
     const root = `${timestamp}__${user.id}__${name}`;
-    const assets = await uploadArchiveContent(file, root);
+    const assets = await uploadArchiveContent(file, root, repository);
     return res.json({ root, assets });
   }
-  const asset = await uploadFile(file, name);
+  const asset = await uploadFile(file, name, repository);
   return res.json(asset);
 }
 
 module.exports = { getUrl, upload };
 
-async function uploadFile(file, name) {
+async function uploadFile(file, name, repository) {
```

Comment:
Consider having repository as the first argument (uploading file for repository) and passing only repository id (minimal info required) :bowtie: 😉